### PR TITLE
refactor: finally rename admin menu to dev menu

### PR DESCRIPTION
**Pull request by pvinis**
### Description

we have been calling this the admin menu, and the dev menu. for the longest time we called it admin menu but we know its wrong, and we call it dev menu but we know people wont find it if we call it that, so we kept calling it the wrong thing.

i am very happy to create this PR and finally rename this to dev menu!

i did *not* change the old native code, so the files there are still called `*admin*`, but the will just go away in time, and that's ok.

along with the rename of the menu, I renamed the `adminOverrides`, since they are not actually admin overrides, they are local overrides on the device. no admin-ing takes place for these.

in the same vein, I moved the `environment` the app is using to the `devicePrefs`, as they are a device-located state. they are not attached to the artsy account in any way, so they should not be in the artsyPrefs. 

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [x] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/store/migration.ts [hunk 1]
```diff
@@ -250,6 +252,16 @@ export const artsyAppMigrations: Migrations = {
...
   },
+  [Versions.RenameAdminToLocalOverridesForFeatures]: (state) => {
+    state.artsyPrefs.features.localOverrides = state.artsyPrefs.features.adminOverrides
+    delete state.artsyPrefs.features.adminOverrides
+  },
+  [Versions.MoveEnvironmentToDevicePrefsAndRenameAdminToLocal]: (state) => {
+    state.devicePrefs.environment = state.artsyPrefs.environment
+    delete state.artsyPrefs.environment
+    state.devicePrefs.environment.localOverrides = state.devicePrefs.environment.adminOverrides
+    delete state.devicePrefs.environment.adminOverrides
```
Comment (MounirDhahri) [inline src/app/store/migration.ts:263]: **thought:** is this needed? I believe it's already done above right?

Comment (pvinis) [reply src/app/store/migration.ts:263]: i am trying to keep as clean a state as possible, so after assigning it, I also delete the old 🤷

File: docs/developing_a_feature.md [hunk 2]
```diff
@@ -101,12 +101,12 @@ This is marked as unsafe because it will not cause react components to re-render
 
 To enable your feature on a physical or virtual device
 
-- Open the admin menu by pressing `command + control + z` on a mac. If this doesn't work:
+- Open the dev menu by pressing `command + control + z` on a mac. If this doesn't work:
```
Comment (patrinoua) [inline docs/developing_a_feature.md:104]: Can we perhaps call it Dev menu / DevMenu / Dev Menu to show it's more like a component, an actual thing?

Comment (patrinoua) [reply docs/developing_a_feature.md:104]: In debugging.md we refer to it as Dev Menu so maybe we can stick to that

File: src/app/store/migration.ts (hunk 1 above)
Comment (MounirDhahri) [reply src/app/store/migration.ts:263]: makes sense. this affects only devs so it's not a problem anyway

File: docs/developing_a_feature.md (hunk 2 above)
Comment (pvinis) [reply docs/developing_a_feature.md]: done

**Review by MounirDhahri — APPROVED** [review]

**Review by gkartalis — APPROVED** [review]
Let's ship this???

**Comment by pvinis** [comment]
i was just waiting for @brainbicycle but i guess we we can yea

**Pull request merged**